codeHF/PlotEfficiency.C

```diff
@@ -51,49 +51,88 @@ Int_t PlotEfficiency(TString pathFile = "AnalysisResults.root", TString particle
 
     TString outputDir = Form("hf-task-%s", particle.Data()); // analysis output directory with histograms
 
+    TString nameHistRec;
+    TString nameHistgen;
+
     // inclusive candidates
-    TString nameHistRec = outputDir + "/hPtRecSig"; // reconstruction level pT of matched candidates
-    //nameHistRec = outputDir + "/hPtGenSig"; // generator level pT of matched candidates (no pT smearing)
-    TString nameHistgen = outputDir + "/hPtGen"; // generator level pT of generated particles
+    if (particles == " lc ") {
+      nameHistRec = "hf-task-lc/MC/reconstructed/signal/hPtRecSig"; // reconstruction level pT of matched candidates
+      nameHistgen = "hf-task-lc/MC/generated/signal/hPtGen";        // generator level pT of generated particles
+    } else {
+      nameHistRec = outputDir + "/hPtRecSig"; // reconstruction level pT of matched candidates
+      // nameHistRec = outputDir + "/hPtGenSig"; // generator level pT of matched candidates (no pT smearing)
+      nameHistgen = outputDir + "/hPtGen"; // generator level pT of generated particles
+    }
+
     TH1F* hPtRecIncl = (TH1F*)file->Get(nameHistRec.Data());
     if (!hPtRecIncl) {
-      Printf("Error: Failed to load %s from %s", nameHistRec.Data(), pathFile.Data());
-      return 1;
+      hPtRecIncl = (TH1F*)file->Get("hf-task-lc/MC/reconstructed/signal/hPtRecSig");
+      if (!hPtRecIncl) {
```

Comment:
Are you not checking the same path twice in case of Lc? You should have the `nameHistRec` set already when trying to load the histogram.

Reply:
Yes, because there were some cases where it did not work and that line was added to avoid the crash. But now this is fixed and the line removed.